src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/Unix/HelpersOpenSsl.cs

```diff
@@ -0,0 +1,203 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Security.Cryptography;
+using System.Security.Cryptography.Pkcs;
+using System.Security.Cryptography.X509Certificates;
+
+using Microsoft.Win32.SafeHandles;
+
+namespace Internal.Cryptography.Pal.OpenSsl
+{
+    internal static class OpenSslHelpers
+    {
+        public static string GetContentType(this SafeCmsHandle cmsHandle)
+        {
+            using (SafeSharedAsn1ObjectHandle oidAsn1 = Interop.Crypto.CmsGetMessageContentType(cmsHandle))
+            {
+                Interop.Crypto.CheckValidOpenSslHandle(oidAsn1);
+                return Interop.Crypto.GetOidValue(oidAsn1);
+            }
+        }
+
+        public static ContentInfo GetEmbeddedContent(this SafeCmsHandle cmsHandle)
+        {
+            string oid;
+            Oid contentType;
+            byte[] content;
+
+            using (SafeSharedAsn1ObjectHandle oidAsn1 = Interop.Crypto.CmsGetEmbeddedContentType(cmsHandle))
+            {
+                Interop.Crypto.CheckValidOpenSslHandle(oidAsn1);
+                oid = Interop.Crypto.GetOidValue(oidAsn1);
+            }
+
+            contentType = new Oid(oid);
+
+            using (SafeSharedAsn1OctetStringHandle encodedContent = Interop.Crypto.CmsGetEmbeddedContent(cmsHandle))
+            {
+                // encodedContent can be a null pointer if there is no content. In this case the content should be set
+                // to an empty byte array. 
+                if (encodedContent.IsInvalid)
+                    content = new byte[0];
```

Comment:
Please use `Array.Empty<byte>()` instead of `new byte[0]`.  The method returns the shared empty array, calling new returns a unique one, which just wastes a little bit of memory.